src/region/regionUtil.ts

```diff
@@ -29,6 +44,98 @@ export const centerRegion = (shape: Shape): Position => {
     };
 };
 
+export const getPoints = (shape: Shape): [Point, Point, Point, Point] => {
+    const { height, width, x, y } = shape;
+
+    // Define the points from x,y and then in a clockwise fashion
+    // p1      p2
+    // +-------+
+    // |       |
+    // +-------+
+    // p4      p3
+    const p1 = { x, y };
+    const p2 = { x: x + width, y };
+    const p3 = { x: x + width, y: y + height };
+    const p4 = { x, y: y + height };
+
+    return [p1, p2, p3, p4];
+};
+
+export function selectTransformationPoint(shape: Shape, rotation: number): Point | null {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [p1, p2, p3, p4] = getPoints(shape);
+
+    // Determine which point will be the new x,y (as defined as the top left point) after rotation
+    // If -90deg: use p2
+    // If -180deg: use p3
+    // If -270deg: use p4
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return p2;
+        case ROTATION_TWICE_DEG:
+            return p3;
+        case ROTATION_THRICE_DEG:
+            return p4;
+        default:
+            return null;
```

Comment:
Should this return `p1`?

Reply:
It could, but returning `null` is more helpful in determining that no transformation needs to take place, otherwise the guard clause in `getTransformedShape` would get more complicated because it would need to check to see if the point returned is the first point

Reply:
Maybe, but utility functions should be fairly generic and not try to implement flow control of their parents. There's also already a check for `const isNoRotation = rotation % 360 === 0;`, so it seems like that case is already covered.

Reply:
Fair enough, I'll update this